robots/cmd/flake-report-creator/cmd/jenkins.go

```diff
@@ -481,19 +487,40 @@ func convertJunitFileDataToReport(junitFilesFromArtifacts []gojenkins.Artifact,
 	return reportsPerJob
 }
 
-func writeReportToFile(startOfReport time.Time, endOfReport time.Time, reports []*flakefinder.JobResult) {
+func writeReportToFile(startOfReport time.Time, endOfReport time.Time, reports []*flakefinder.JobResult, outputFile string) {
 	parameters := flakefinder.CreateFlakeReportData(reports, []int{}, endOfReport, "kubevirt", "kubevirt", startOfReport)
+	jLog.Printf("writing output to %s", outputFile)
 
-	jLog.Printf("writing output to %s", globalOpts.outputFile)
+	writeHTMLReportToOutputFile(outputFile, JenkinsReportTemplate, JenkinsReportParams{Params: parameters, JenkinsBaseURL: opts.endpoint})
 
-	reportOutputWriter, err := os.OpenFile(globalOpts.outputFile, os.O_CREATE|os.O_WRONLY, 0644)
-	if err != nil && err != os.ErrNotExist {
+	writeJSONToOutputFile(strings.TrimSuffix(outputFile, ".html")+".json", parameters)
+}
+
+func writeHTMLReportToOutputFile(outputFile string, reportTemplate string, params interface{}) {
+	reportOutputWriter, err := createReportOutputWriter(outputFile)
+	defer reportOutputWriter.Close()
+
+	err = flakefinder.WriteTemplateToOutput(reportTemplate, params, reportOutputWriter)
+	if err != nil {
 		jLog.Fatalf("failed to write report: %v", err)
 	}
+}
+
+func writeJSONToOutputFile(jsonOutputFile string, parameters flakefinder.Params) {
+	reportOutputWriter, err := createReportOutputWriter(jsonOutputFile)
 	defer reportOutputWriter.Close()
 
-	err = flakefinder.WriteTemplateToOutput(JenkinsReportTemplate, JenkinsReportParams{Params: parameters, JenkinsBaseURL: opts.endpoint}, reportOutputWriter)
+	encoder := json.NewEncoder(reportOutputWriter)
+	err = encoder.Encode(parameters.Data)
 	if err != nil {
 		jLog.Fatalf("failed to write report: %v", err)
 	}
 }
+
+func createReportOutputWriter(outputFile string) (*os.File, error) {
+	reportOutputWriter, err := os.OpenFile(outputFile, os.O_CREATE|os.O_WRONLY, 0644)
```

Comment:
QQ - Does this require 0644 permissions or would it work with 0600?

Reply:
Good question: I haven't checked, but since the script runs the tool inside a container with [a host mount attached to `/tmp`](https://github.com/kubevirt/project-infra/blob/main/hack/flake-report-creator.sh#L7) I am not sure what happens TBH